src/Workspaces/Core/Desktop/Workspace/Host/Documentation/DocumentationProviderServiceFactory.cs

```diff
@@ -29,16 +30,53 @@ public DocumentationProvider GetDocumentationProvider(string assemblyPath)
                     throw new ArgumentNullException(nameof(assemblyPath));
                 }
 
-                assemblyPath = Path.ChangeExtension(assemblyPath, "xml");
+                string xmlDocumentationFilePath;
+                if (!TryFindDocumentationFile(assemblyPath, out xmlDocumentationFilePath))
+                {
+                    return DocumentationProvider.Default;
```

Comment:
Should I be caching this NullDocumentationProvider in _assemblyPathToDocumentationProviderMap and return that if we couldnt resolve to a doc file on disk?